### PR TITLE
Updating stars when deleting review

### PR DESCRIPTION
**Pull request by katiacammisa**
# Description

Updating stars when creating and deleting review

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How has this been tested?

Please describe the tests that you ran to verify your changes.

- [X] Automation testing (Unit or integration tests)
- [X] Manual testing (Postman)

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] My pull request is up to date with develop

### CODE REVIEW

File: src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java [hunk 1]
```diff
@@ -44,24 +46,27 @@ public Review find(Long id) {
...
     @Override
     public Review save(Review review) {
-        review.setBook(bookService.updateStars(review.getBook().getId(), review.getStars()));
+        review.setBook(bookService.calculateStars(review.getBook().getId(), review.getStars(), Strategy.CREATE));
         return repository.save(review);
     }
 
     @Override
     public Review update(Long id, Review review) {
+        Book updated_book = bookService.calculateStars(find(id).getBook().getId(), review.getStars(), Strategy.UPDATE, find(id).getStars());
```
Comment (brianfroschauer) [inline src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java:55]: por que usamos snake case en lugar de camel case?

File: src/main/java/com/austral/bookin/service/book/BookServiceImpl.java [hunk 2]
```diff
@@ -61,14 +62,23 @@ public Book save(Book book, MultipartFile file) {
     }
 
     @Override
-    public Book updateStars(long id, int stars) {
+    public Book calculateStars(long id, int stars, Strategy strategy, int... old_stars) {
```
Comment (brianfroschauer) [inline src/main/java/com/austral/bookin/service/book/BookServiceImpl.java:65]: camel case -> oldStars

**Review by brianfroschauer — CHANGES_REQUESTED** [review]

File: src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java (hunk 1 above)
Comment (katiacammisa) [reply src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java:55]: Se me mezclan los lenguajes, en scala lo usamos asi

File: src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java [hunk 3]
```diff
@@ -44,24 +46,27 @@ public Review find(Long id) {
...
                     if (review.getComment() != null) old.setComment(review.getComment());
+                    old.setBook(updated_book);
                     return repository.save(old);
                 })
                 .orElseThrow(NotFoundException::new);
     }
 
     @Override
     public void delete(Long id) {
+        bookService.calculateStars(find(id).getBook().getId(), find(id).getStars(), Strategy.DELETE);
```
Comment (brianfroschauer) [inline src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java:69]: para qué actualizamos las estrellas en un libro que inmediatamente despues vamos a eliminar?

**Review by brianfroschauer — CHANGES_REQUESTED** [review]

File: src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java (hunk 3 above)
Comment (katiacammisa) [reply src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java:69]: Ahi estamos borrando la review, no el libro

File: src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java (hunk 1 above)
Comment (katiacammisa) [reply src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java]: Ahi quedó arreglado

File: src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java (hunk 3 above)
Comment (diegobaldassare) [reply src/main/java/com/austral/bookin/service/review/ReviewServiceImpl.java:69]: Faltan actualizar las estrellas de un libro cuando se modifica una reseña.

**Pull request merged**